mitodl/mit-open
# Featured API endpoint

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/4183

### Description (What does it do?)
- Adds an API endpoint for "featured" resources that queries for all resources in the "featured_list" of each of 5 offeror channels, sorted by the position of the resource in each featured list.  The endpoint allows for the same filtering options as the learning_resources api endpoint.
- Adds a management command (for dev purposes only) that auto-generates a featured list for each of the 5 offerors and populates it with x number of published courses from the relevant offeror.


### How can this be tested?
- Run the necessary management commands if necessary to get courses from each of the 5 offerors (adding required .env values too)
  ```
  backpopulate_prolearn_data
  backpopulate_ocw_data --skip-contentfiles
  backpopulate_xpro_data
  backpopulate_mitxonline_data
  ```
- Run `backpopulate_resource_channels`
- Run `populate_featured_lists`
- Go to http://localhost:8063/api/v1/featured/, there should be a total of 50 results (10 from each of 5 offerors).  
- Check that the result ordering coincides with the position of the resource from its respective featured learning path (you can see those at http://localhost:8063/learningpaths).  So the first 5 results should be the 1st resource from each of the 5 featured lists, the next 5 results should be the 2nd resource from each of the lists, etc.
- Try out some filtering options like `?certification=true`, `professional=true`, `free=true`.  The results should be filtered appropriately.
